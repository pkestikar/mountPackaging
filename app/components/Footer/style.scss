.mp--footer {
  width: 100%;
  background: var(--dark-blue);
  color: white;
  &--inner {
    padding: 5rem 0;
    @media only screen and (max-width: $medium) {
      padding: 2rem 0;
    }
  }
  .footer-logo-wrapper {
    display: flex;
    justify-content: center;
  }
  .footer-nav-wrapper {
    display: flex;
    @media only screen and (max-width: $medium) {
      flex-direction: column;
      align-items: center;
      border-top: 1px solid white;
      border-bottom: 1px solid white;
      margin: 1rem;
      padding-bottom: 1rem;
    }
    a {
      width: 7rem;
      text-align: center;
      text-decoration: none;
      color: white;
      margin-top: 1rem;
    }
    @media only screen and (max-width: $medium) {
      a {
        width: 6rem;
      }
    }
    a:hover {
      text-decoration: underline;
    }
    a:active {
      text-decoration: underline;
    }
  }

  .footer-contact-info {
    .footer-connect-heading {
      font-size: 1.2rem;
      width: 100%;
      text-align: center;
    }
    .footer-address {
      line-height: 1.7;
      padding: 1rem 0;
      @media only screen and (max-width: $medium) {
        text-align: center;
      }
    }
    .footer-email {
      @media only screen and (max-width: $medium) {
        text-align: center;
      }
    }
  }

  .mp--copyright-notice {
    padding-bottom: 2rem;
    text-align: center;
  }
}
