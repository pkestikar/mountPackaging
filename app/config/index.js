export const IMAGE_PATH = '/images/';
export const NAV_ITEMS = [
    {
        title: 'About Us',
        link: '/about-us/'
    },
    {
        title: 'Work',
        link: '/work/'
    },
    {
        title: 'Services',
        link: '/services/'
    },
    {
        title: 'Contact',
        link: '/contact/'
    },
];
export const CONTACT_EMAIL = `mountindia@mountindia.net`;
export const COPYRIGHT_NOTICE_TEXT = `Copyright 2020 Mount Packaging Machinery Pvt Ltd.`;

export const WORK_PAGE_INDIVIDUAL_UNIT_NAMES = [
    'Bag in box filling machines',
    'Box case erectors and sealers',
    'Cap elevators',
    'Capping machines',
    'Container and cap unscramblers',
    'De-Palletiser',
    'Gear pump filling machines',
    'Gravimetric filling machine',
    'Infeed Table',
    'Inline check-weighing',
    'Labelling system(s)',
    'Lidder (pick and place or sweep on)',
    'Misprint rejection system for cans and boxes',
    'Palletiser',
    'Pantograph filling machine',
    'Peristaltic pump filling machines',
    'Printing and ink jet coding',
    'Roller conveyor systems (automated or gravity fed)',
    'Volumetric pump filling machines',
    'Weigh fill machines'
];
export const WORK_PAGE_MACHINE_SHOWCASE = [
    {
        title: 'Semi Automatic Bulk Filling Machine',
        image: 'semi-automatic-bulk-filling.png'
    },
    {
        title: 'Semi Automatic PVF-5 With Lid Press Machine',
        image: 'semi-automatic-pvf-5.png'
    },
    {
        title: 'Semi Automatic Bulk & Retail Combo Filling Machine',
        image: 'semi-automatic-bulk-and-retail.png'
    },
    {
        title: 'Semi Automatic Twin PVF-5 Filling With Lid Press Machine',
        image: 'semi-automatic-twin-pvf-5.png'
    },
    {
        title: 'Semi Automatic PVF-20 Filling Machine - Type 1',
        image: 'semi-automatic-pvf-20-1.png'
    },
    {
        title: 'Semi Automatic PVF-20 Filling Machine - Type 2',
        image: 'semi-automatic-pvf-20-2.png'
    },
    {
        title: 'Fully Automatic Retail Filling Machine',
        image: 'fully-automatic-retail-filling.png'
    },
    {
        title: 'Fully Automatic Bulk Filling Machine',
        image: 'fully-automatic-bulk-filling.png'
    },
    {
        title: 'Semi Automatic Retail Line With Lid Place & Press Filling Machine',
        image: 'semi-automatic-retail-line.png'
    }
];

export const SERVICES_PAGE_SERVICE_LIST = [
    {
        title: 'Installation',
        description: 'Our engineers will make sure the machinery is properly installed at the production site of the customer.',
        image: 'installation.svg'
    },
    {
        title: 'Maintenance',
        description: 'We provide regular maintenance services to have the production line keep going smoothly.',
        image: 'maintenance.svg'
    },
    {
        title: 'Spare parts supply',
        description: 'We have a large inventory of consumable stores, ready to dispatch as early possible.',
        image: 'spare-parts-supply.svg'
    },
    {
        title: 'Modifications/Upgrades',
        description: 'Our machines are designed to adapt the modifications/upgrades whenever required.',
        image: 'upgrade.svg'
    }
]