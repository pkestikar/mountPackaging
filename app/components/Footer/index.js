import React from 'react';
import Container from '@material-ui/core/Container';
import Grid from '@material-ui/core/Grid';
import { IMAGE_PATH, NAV_ITEMS, CONTACT_EMAIL, COPYRIGHT_NOTICE_TEXT } from "../../config";
import {NavLink} from "react-router-dom";

const Footer = (props) => {
    return (
        <div className={'mp--footer'}>
            <Container disableGutters>
                <Grid container className={'mp--footer--inner'}>
                    <Grid item className={'footer-logo-wrapper'} xs={12} md={2}>
                        <div>
                            <img src={`${IMAGE_PATH}mount-logo-light.png`} alt="Logo"/>
                        </div>
                    </Grid>
                    <Grid item className={'footer-nav-wrapper'} xs={12} md={6}>
                        {
                            NAV_ITEMS.map((item, key) => {
                                return (
                                    <NavLink key={`nav-item-${key}`} to={`${item.link}`} activeClassName={'active'}>{item.title}</NavLink>
                                )
                            })
                        }
                    </Grid>
                    <Grid item className={'footer-contact-info'} xs={12} md={4}>
                        <div className={'footer-connect-heading'}>LET'S CONNECT!</div>
                        <div className="footer-address">
                            We are located at:<br />
                            Mahatma Jyotirao Phule Marg, <br />
                            Dhadge Industrial Estate, Nanded Phata, <br />
                            Sinhgad Road, <br />
                            Pune 411041, <br />
                            Maharashtra, India. <br />
                        </div>
                        <div className="footer-email">
                            Email: {CONTACT_EMAIL}
                        </div>
                    </Grid>
                </Grid>

                <div className="mp--copyright-notice">
                    {COPYRIGHT_NOTICE_TEXT}
                </div>
            </Container>
        </div>
    );
};

export default Footer;
