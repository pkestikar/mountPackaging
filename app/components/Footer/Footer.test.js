import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import {CONTACT_EMAIL, COPYRIGHT_NOTICE_TEXT} from "../../config";

import Footer from './index';

describe('Footer component', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<Footer />);
    });

    it('should render correctly', () => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should have the right logo (mount light)', () => {
        expect(wrapper.find('img').prop('src')).toContain('mount-logo-light.png');
    });

    it('should contain the right copyright notice text', () => {
        expect(wrapper.find('.mp--copyright-notice').text()).toEqual(COPYRIGHT_NOTICE_TEXT);
    });

    it('should have the right email address from config', () => {
        expect(wrapper.find('.footer-email').text()).toContain(CONTACT_EMAIL);
    });

    it('should have the right number of the footer nav links', () => {
        expect(wrapper.find('.footer-nav-wrapper').children().length).toEqual(4);
    });
});
