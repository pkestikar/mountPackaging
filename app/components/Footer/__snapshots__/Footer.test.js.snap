// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should render correctly 1`] = `
<div
  className="mp--footer"
>
  <WithStyles(ForwardRef(Container))
    disableGutters={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="mp--footer--inner"
      container={true}
    >
      <WithStyles(ForwardRef(Grid))
        className="footer-logo-wrapper"
        item={true}
        md={2}
        xs={12}
      >
        <div>
          <img
            alt="Logo"
            src="/images/mount-logo-light.png"
          />
        </div>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="footer-nav-wrapper"
        item={true}
        md={6}
        xs={12}
      >
        <NavLink
          activeClassName="active"
          key="nav-item-0"
          to="/about-us/"
        >
          About Us
        </NavLink>
        <NavLink
          activeClassName="active"
          key="nav-item-1"
          to="/work/"
        >
          Work
        </NavLink>
        <NavLink
          activeClassName="active"
          key="nav-item-2"
          to="/services/"
        >
          Services
        </NavLink>
        <NavLink
          activeClassName="active"
          key="nav-item-3"
          to="/contact/"
        >
          Contact
        </NavLink>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        className="footer-contact-info"
        item={true}
        md={4}
        xs={12}
      >
        <div
          className="footer-connect-heading"
        >
          LET'S CONNECT!
        </div>
        <div
          className="footer-address"
        >
          We are located at:
          <br />
          Mahatma Jyotirao Phule Marg, 
          <br />
          Dhadge Industrial Estate, Nanded Phata, 
          <br />
          Sinhgad Road, 
          <br />
          Pune 411041, 
          <br />
          Maharashtra, India. 
          <br />
        </div>
        <div
          className="footer-email"
        >
          Email: 
          mountindia@mountindia.net
        </div>
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
    <div
      className="mp--copyright-notice"
    >
      Copyright 2020 Mount Packaging Machinery Pvt Ltd.
    </div>
  </WithStyles(ForwardRef(Container))>
</div>
`;
